backend/src/packages/articles/article.entity.ts

```diff
@@ -0,0 +1,96 @@
+import { type IEntity } from '~/libs/interfaces/interfaces.js';
+import { type WithNullableKeys } from '~/libs/types/types.js';
+
+import { type ArticleEntityType } from './libs/types/types.js';
+
+class ArticleEntity implements IEntity {
+  private 'id': number | null;
+  private 'title': string;
+  private 'text': string;
+  private 'userId': number;
+  private 'promptId': number | null;
+  private 'genreId': number;
+  private 'publishedAt': string | null;
+
+  private constructor({
+    id,
+    title,
+    text,
+    userId,
+    promptId,
+    genreId,
+    publishedAt,
+  }: WithNullableKeys<ArticleEntityType, 'id'>) {
+    this.id = id;
+    this.title = title;
+    this.text = text;
+    this.userId = userId;
+    this.promptId = promptId;
+    this.genreId = genreId;
+    this.publishedAt = publishedAt;
+  }
+
+  public static initialize({
+    id,
+    title,
+    text,
+    userId,
+    promptId,
+    genreId,
+    publishedAt,
+  }: ArticleEntityType): ArticleEntity {
+    return new ArticleEntity({
+      id,
+      title,
+      text,
+      userId,
+      promptId,
+      genreId,
+      publishedAt,
+    });
+  }
```

Comment:
I think, we can simplify this as this destructurization just duplicates fields
```suggestion
  public static initialize(articleEntity: ArticleEntityType): ArticleEntity {
    return new ArticleEntity(articleEntity);
  }
```

Reply:
Also, check other places in your code where this suggestion is applicable. @andanf-e @nickolay-grechkin what to you think?

Reply:
I think I can do that now. And when ArticleEntity expands, we may think about what to change.

Reply:
When I was doing it, I was looking at UserEntity as an example because it was in the starter

Reply:
Well, the main idea is that the ArticleEntityType itself can be wider, but I only need these fields in the output

Reply:
Yep @drMangust you're right - because we may have wider data shape - we may form it only with needed things

Reply:
@andanf-e So do we leave it as it is?

Reply:
Oh, Okay then, agreed